### DISCUSSION TITLE
Confusing interface for "New folder from Git"

### DISCUSSION THREAD

**Discussion by etiennebacher** [posit-dev/positron#4928] [Feedback]
I like the "New folder from Git" button, but I think it could be a bit clearer. Currently, when I click on "OK" after putting the link to my Git repo:

1. the modal doesn't disappear and instead there's a "toast" notification on the bottom right indicating the progress of the "git clone" process. This is hard to see because the first modal is still highlighted. 
1. the "Cancel" button in the modal doesn't actually cancel the "git clone" process once it has started, this can only be done with the "Cancel" button in the "toast" notification.
1. the "OK" button is not disabled, meaning that if I click several times then it spawns several "git clone" processes (I suppose, since there are several "toast" notifications). Combined with the first point, it's easy to mistakenly trigger several of those processes. 

IMO, a clearer interface would be to show the progress of "git clone" in the same modal, below the "OK" and "Cancel" buttons. This way, we would clearly see that the process started, we would be able to use the unique "Cancel" button, and it would allow disabling the "OK" button if a process is already running. Just my 2 cents of course :-)

Edit: On the other hand, that would block the user from doing something else until "git clone" finishes, which can take a while. An easier solution might be to just close the modal once "OK" is clicked.

**Comment by timtmok** [comment]
I think some kind of updating modal would make sense here. Dismissing the modal to run the action in the background and allowing the user to do something else while cloning the repo would require some kind of interrupt once the clone is done. Otherwise, the IDE will immediately open the newly cloned repo and that would feel disorienting.

The action itself is meant to clone and open the repo. I think it's useful that opening the folder is immediate and there probably isn't anything else to do while waiting to clone the repo. Maybe the user wants to open the repo in a new window?

**Reply by etiennebacher** [reply]
> The action itself is meant to clone and open the repo. I think it's useful that opening the folder is immediate and there probably isn't anything else to do while waiting to clone the repo. Maybe the user wants to open the repo in a new window?

Oh right, I forgot to mention that I almost always tick the "Open in a new window", so for me the workflow is like "Clone this repo in the background and open it in a new folder as soon as it's finished, but let me do some other stuff on this folder in the meantime"

I think putting the progress bar in the same modal makes sense if the user wants to replace the current folder with the cloned one, not if they want to open in a separate window.

**Comment by juliasilge** [comment]
Thank you for this feedback! 🙌 I agree this experience could be improved. Let's use this discussion here to track folks' ideas and interest.
